Final Project/Server/fiimentor/controllers/professorController.ts

```diff
@@ -0,0 +1,52 @@
+import { ProfessorRepository } from "../Repositories/ProfessorRepository";
+import { UserRepository } from "../Repositories/UserRepository";
+import HttpStatus from "http-status-codes";
+
+interface ProfessorPayload {
+    id : number ,
+    UserId : number ,
+    FirstName : string,
+    LastName : string 
+}
+
+async function professor_list (req : any , res: any) {
+
+    try {
+
+        let proffesorsArray = new Array<ProfessorPayload>();
+
+        const userRepository = new UserRepository();
+        const users = await userRepository.getAll();
+
+    
+        const professorRepository = new ProfessorRepository();
+        const allProfessors = await professorRepository.getAll();
+
+
+        let payload : ProfessorPayload;
+        let contor : number;
+        allProfessors.forEach(item => {
+            
+            contor = item.userId - 1;
```

Comment:
poti sa i spui index, suna mai sugestiv si il poti declara direct aici cu const, nu in afara forEach ului